File: code/00-pull-historical-load-data.R
library(gdata)
library(RCurl)
library(XML)
library(XLConnect)


# References
# http://gastonsanchez.com/work/webdata/getting_web_data_r2_reading_files.pdf

# URL locations
# http://www.pjm.com/pub/operations/hist-meter-load/2015-hourly-loads.xls
# http://www.pjm.com/pub/operations/hist-meter-load/2014-hourly-loads.xls



#using GDATA 
load15_home = "http://www.pjm.com/pub/operations/hist-meter-load/2015-hourly-loads.xls"
load14_home = "http://www.pjm.com/pub/operations/hist-meter-load/2014-hourly-loads.xls"

#sheetCount(data_home)

perl = "C:/Users/jonfar/AppData/Local/Programs/Git/bin/perl.exe"

write.csv(read.xls(data_home, sheet = 9, perl=perl), file="C:/Users/jonfar/Documents/Research/comp-2015/data/rawdat/load15.csv") 


dat = read.xls(load15_home, sheet = 9, perl=perl)

head(data)
summary(data)
names(data)

File: code/00-pull-historical-load-data.py
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 20 2015

@author: jonfar
"""

#!/usr/bin/env python
# -*- coding: utf-8 -*-


import requests
import urllib

load15 = "http://www.pjm.com/pub/operations/hist-meter-load/2015-hourly-loads.xls"
urllib.urlretrieve(load15, "load15.xls")

load14 = "http://www.pjm.com/pub/operations/hist-meter-load/2014-hourly-loads.xls"
urllib.urlretrieve(load14, "load14.xls")

load13 = "http://www.pjm.com/pub/operations/hist-meter-load/2013-hourly-loads.xls"
urllib.urlretrieve(load13, "load13.xls")

load12 = "http://www.pjm.com/pub/operations/hist-meter-load/2012-hourly-loads.xls"
urllib.urlretrieve(load12, "load12.xls")

load11 = "http://www.pjm.com/pub/operations/hist-meter-load/2011-hourly-loads.xls"
urllib.urlretrieve(load11, "load11.xls")

